lib/lte_link_control/lte_lc.c

```diff
@@ -630,6 +957,262 @@ static int parse_nw_reg_status(const char *at_response,
 	return err;
 }
 
+/**@brief Parses an AT command response, and returns the current RRC mode.
+ *
+ * @param at_response Pointer to buffer with AT response.
+ * @param mode Pointer to where the RRC mode is stored.
+ * @param mode_index Parameter index for mode.
+ *
+ * @return Zero on success or (negative) error code otherwise.
+ */
+static int parse_rrc_mode(const char *at_response,
+			  enum lte_lc_rrc_mode *mode,
+			  size_t mode_index)
+{
+	int err, temp_mode;
+	struct at_param_list resp_list = {0};
+
+	if ((at_response == NULL) || (mode == NULL)) {
```

Comment:
This cannot happen. You should remove those checks from static functions. You already check the validity of `at_response` in the caller function.